### PR TITLE
Documentation: Add info on `searchpromotions` stored queries

### PR DESCRIPTION
**Pull request by sgfost**
**This PR adds some additional information about managing stored search queries that were moved to the `contrib.searchpromotions` app in 5.0.** I didn't find much info on the docs when I ran into issues with migrating, so the intent here is to help clear up any potential confusion about the migration and general usage.

## Changes

- Add brief explanation and usage example for the `Query` and `QueryDailyHits` models to the search promotions page
- Add note regarding the new cleanup command and link to the updated page in 5.0 release notes
- Redirect to `searchpromotions` where this functionality is referenced

## Checks

-   [x] Do the tests still pass?[^1]
-   [x] Does the code comply with the style guide?
-   [x] For Python changes: Have you added tests to cover the new/fixed behaviour? **N/A**
-   [x] For front-end changes: Did you test on all of Wagtail’s supported environments?[^2] **N/A**
-   [x] For new features: Has the documentation been updated accordingly?

**Please describe additional details for testing this change**.

[^1]: [Development Testing](https://docs.wagtail.org/en/latest/contributing/developing.html#testing)
[^2]: [Browser and device support](https://docs.wagtail.org/en/latest/contributing/developing.html#browser-and-device-support)
[^3]: [Accessibility Target](https://docs.wagtail.org/en/latest/contributing/developing.html#accessibility-targets)

### CODE REVIEW

File: docs/reference/contrib/searchpromotions.md [hunk 1]
```diff
@@ -10,6 +10,8 @@ The `searchpromotions` module provides the models and user interface for managin
 
 "Promoted search results" allow editors to explicitly link relevant content to search terms, so results pages can contain curated content in addition to results from the search engine.
 
+As of Wagtail 5.0, the `searchpromotions` module contains the `Query` and `QueryDailyHits` models used to log search queries.
```
Comment (lb-) [inline docs/reference/contrib/searchpromotions.md:13]:
```suggestion
    ```{versionadded} 5.0
    The `searchpromotions` module contains the `Query` and `QueryDailyHits` models used to log search queries.
    ```
```

Just a drive-by - we can use the built in admonitions for this kind of thing.

**Comment by lb-** [comment]
Thanks @sgfost  has been flagged as needs review

File: docs/reference/contrib/searchpromotions.md (hunk 1 above)
Comment (laymonage) [reply docs/reference/contrib/searchpromotions.md:13]: I don't think this paragraph is necessary, the first paragraph in the added "Managing stored search queries" section below should be sufficient.

File: docs/reference/contrib/searchpromotions.md [hunk 2]
```diff
@@ -50,3 +52,58 @@ To retrieve a list of promoted search results for a particular search query, you
...
+
+(searchpromotions_garbage_collect)=
+
+### `searchpromotions_garbage_collect`
+
+```sh
+./manage.py searchpromotions_garbage_collect
+```
+
+On high traffic websites, the stored queries and daily hits logs may get large and you may want to clean out old records. This command cleans out all search query logs that are more than one week old (or a number of days configurable through the [`WAGTAILSEARCH_HITS_MAX_AGE`](wagtailsearch_hits_max_age) setting).
```
Comment (laymonage) [inline docs/reference/contrib/searchpromotions.md:109]: We should probably add a `versionadded` (`versionchanged`?) here to note the original command name

File: docs/reference/management_commands.md [hunk 3]
```diff
@@ -152,6 +150,10 @@ Displays a summary of the contents of the references index. This shows the numbe
 
 Wagtail keeps a log of search queries that are popular on your website. On high traffic websites, this log may get big and you may want to clean out old search queries. This command cleans out all search query logs that are more than one week old (or a number of days configurable through the [`WAGTAILSEARCH_HITS_MAX_AGE`](wagtailsearch_hits_max_age) setting).
 
+```{versionchanged} 5.0
+The `Query` model has been moved to the [searchpromotions](editors_picks) module and this functionality can now be invoked with the [](searchpromotions_garbage_collect) command.
```
Comment (laymonage) [inline docs/reference/management_commands.md:154]: I think it's better to mention the functionality itself rather than specifically mentioning the `Query` model here. Also, we should probably replace the description pointing to the new command entirely, with link to the release notes about the migration.

**Review by laymonage — COMMENTED** [review]
Thank you, I think this is worth getting in before the 5.1 RC, as we'll need to remove any 5.0 admonitions in 5.2

File: docs/reference/contrib/searchpromotions.md [hunk 4]
```diff
@@ -50,3 +52,58 @@ To retrieve a list of promoted search results for a particular search query, you
...
+
+#### `Query.get()`
+
+Retrieves a stored search query or creates a new one if it doesn't exist.
+
+#### `add_hit()`
+
+Records another daily hit for a search query by creating a new record or incrementing the number of hits for an existing record. Defaults to using the current date but an optional `date` parameter can be passed in.
+
+#### Example search view
```
Comment (laymonage) [inline docs/reference/contrib/searchpromotions.md:68]: The original content in `wagtail.search`'s documentation should be removed, and the accompanying template should be put here as well.

File: docs/reference/contrib/searchpromotions.md [hunk 5]
```diff
@@ -50,3 +52,58 @@ To retrieve a list of promoted search results for a particular search query, you
...
+
+### Managing stored search queries
+
+The `searchpromotions` module keeps a log of search queries as well as the number of daily hits through the `Query` and `QueryDailyHits` models.
+
+#### `Query.get()`
+
+Retrieves a stored search query or creates a new one if it doesn't exist.
+
+#### `add_hit()`
```
Comment (laymonage) [inline docs/reference/contrib/searchpromotions.md:64]: This should be formatted as a reference documentation I think

**Comment by laymonage** [comment]
Thank you for making your first contribution to Wagtail @sgfost! I've credited you as Scott Foster (found via https://www.comses.net/users/3587/), please let me know if you'd like a different name listed.

**Comment by sgfost** [comment]
Great, thanks for the edits @laymonage
